components/sensor/mcp3204.rst

```diff
@@ -0,0 +1,80 @@
+MCP3204 4-Channel 12-Bit A/D Converter
+======================================
+
+.. seo::
+    :description: Instructions for setting up MCP3204 12-Bit Analog to Digital Converter in ESPHome.
+    :keywords: MCP3204
+    :image: mcp3204.jpg
+
+The Microchip Technology Inc. MCP3204
+devices are successive approximation 12-bit Analog-to-Digital (A/D) converters with on-board sample and
+hold circuitry.
+
+.. figure:: images/mcp3204.jpg
+    :align: center
+    :width: 50.0%
+
+.. _mcp3204-component:
+
+Component/Hub
+-------------
+
+The MCP3204 component allows you to use MCP3204 4-Channel 12-Bit A/D Converter
+(`datasheet <https://ww1.microchip.com/downloads/en/DeviceDoc/21298e.pdf>`__) in ESPHome.
+It uses the :ref:`SPI Bus <spi>` for communication.
+
+Once configured, you can use any of the 4 pins as
+sensors for your projects.
+
+Each pin will respond with a voltage calculated off of the reference_voltage (default is 3.3v).
+It calculates the voltage by multiplying the reference_voltage * value on the pin (basically the percentage of VREF)
+
+Most configurations will set the reference_voltage = VREF (pin 13 on the chip)
+
+If you want just the scaled value you can use the read_data function
+  ``float MCP3204::read_data(uint8_t pin)``
+
```

Comment:
I just realised I removed this functionality by multiplying the value by ref voltage in the parent, happy for the calculation to move back to ``sample()`` if you want?
```suggestion
```